jerry-core/ecma/base/ecma-globals.h

```diff
@@ -257,6 +257,8 @@ typedef enum
 typedef enum
 {
   ECMA_PROPERTY_TYPE_DELETED, /**< deleted property */
+  ECMA_PROPERTY_TYPE_GENERIC = ECMA_PROPERTY_TYPE_DELETED, /**< generic property type
```

Comment:
I'm a bit concerned about polluting this public enum type with `ECMA_PROPERTY_TYPE_GENERIC`, as that marker is only used in and meaningful for `ecma_op_general_object_define_own_property`. What about adding
```c
#define ECMA_PROPERTY_TYPE_GENERIC ECMA_PROPERTY_TYPE_DELETED
```
and
```c
#undef ECMA_PROPERTY_TYPE_GENERIC
```
around `ecma_op_general_object_define_own_property` in `ecma-objects-general.c`?